pkg/kubelet/kubelet.go

```diff
@@ -745,7 +746,26 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	// check node capabilities since the mount path is not the default
 	if len(kubeCfg.ExperimentalMounterPath) != 0 {
 		kubeCfg.ExperimentalCheckNodeCapabilitiesBeforeMount = false
+		resolvePath := filepath.Join(strings.TrimSuffix(kubeCfg.ExperimentalMounterPath, "/mounter"), "rootfs", "etc", "resolv.conf")
+		fileHandle, err := os.OpenFile(resolvePath, os.O_APPEND|os.O_WRONLY, 0600)
+		if err != nil {
+			glog.Errorf("Failed to open file %s, with error %v", resolvePath, err)
+		} else {
+			defer fileHandle.Close()
+
+			dnsString := ""
+			for _, dns := range klet.clusterDNS {
+				dnsString = dnsString + fmt.Sprintf("nameserver %s\n", dns)
+			}
+			dnsString = dnsString + fmt.Sprintf("options rotate\n")
```

Comment:
fmt.Sprintf("options rotate\n" -> "options rotate\n"?

Reply:
what is the difference?

Reply:
We shouldn't need `fmt.Sprintf()` because there is no arguments?

Reply:
Humm..BTW do we really need `"options rotate"`? It won't hurt to do DNS query in the order as defined in resolv.conf right?

Reply:
Without rotate option, the name still cannot be resolved.

Reply:
address the comments, PTAL

Reply:
Yep I think it occasionally worked with `"options rotate"` because sometime kube-dns got rotated as the first server.